pkg/sources/git/git.go

```diff
@@ -616,3 +634,49 @@ func getSafeRemoteURL(repo *git.Repository, preferred string) string {
 	}
 	return safeURL
 }
+
+func handleBinary(repo *git.Repository, chunksChan chan (*sources.Chunk), chunkSkel *sources.Chunk, commitHash plumbing.Hash, path string) error {
+	log.WithField("path", path).Trace("Binary file found in repository.")
+	commit, err := repo.CommitObject(commitHash)
+	if err != nil {
+		return err
+	}
+
+	file, err := commit.File(path)
+	if err != nil {
+		return err
+	}
+
+	reader, err := file.Reader()
+	if err != nil {
+		return err
+	}
+	handled := handlers.HandleFile(reader, chunkSkel, chunksChan)
+
+	if !handled {
+		log.WithField("path", path).Trace("Binary file is not recognized by file handlers. Chunking raw.")
+		chunkSize := 10 * 1024
```

Comment:
nit: might be worth pulling this into a constant for a bit more clarity.
const bufferSize = 10 * 1024

aside: Might be worth down the road getting this logic into a more central place. I notice we have this calculation littered across the codebase. Some places we use a const others we don't.

Reply:
Funny you should mention that #703